presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -255,10 +254,10 @@ public Object evaluate()
         return visitor.process(expression, new NoPagePositionContext());
     }
 
-    public Object evaluate(int position, Page page)
+    public Object evaluate(SymbolResolver inputs)
```

Comment:
I don't understand this class that well so someone please help explain:

after this change, what's the difference between this `evaluate` interface and the `optimize` one?

Reply:
evaluate is to evaluate the expression into object (it is only used in test).
optimize is dead code we should remove.